x-pack/legacy/plugins/lens/public/indexpattern_plugin/datapanel.tsx

```diff
@@ -459,3 +479,38 @@ export const InnerIndexPatternDataPanel = function InnerIndexPatternDataPanel({
 };
 
 export const MemoizedDataPanel = memo(InnerIndexPatternDataPanel);
+
+function Loader(props: { load: () => Promise<unknown>; loadDeps: unknown[] }) {
```

Comment:
Yeah. I thought we might be able to reuse it, but didn't bother with its own file. Done now, though.

As for the abort, I agree, but I think it's too complex to warrant for this use case. It would require rewiring a number of functions which are currently async, so return Promises rather than abortable interfaces.